Produlytics/api-rilevazioni/src/main/java/it/deltax/produlytics/api/detections/adapters/DetectionsAdapter.java

```diff
@@ -146,22 +147,36 @@ public void insertDetection(Detection detection) {
    */
   @Override
   public LimitsInfo findLimits(CharacteristicId characteristicId) {
-    LimitsEntity limitsEntity =
-        this.characteristicRepository.findLimits(
+    CharacteristicEntity characteristicEntity =
+        this.characteristicRepository.findByDeviceIdAndId(
             characteristicId.deviceId(), characteristicId.characteristicId());
 
     Optional<TechnicalLimits> technicalLimits = Optional.empty();
-    if (limitsEntity.getTechnicalLowerLimit().isPresent()
-        && limitsEntity.getTechnicalUpperLimit().isPresent()) {
-      double lowerLimit = limitsEntity.getTechnicalLowerLimit().get();
-      double upperLimit = limitsEntity.getTechnicalUpperLimit().get();
+    if (characteristicEntity.getLowerLimit() != null
+        && characteristicEntity.getUpperLimit() != null) {
+      double lowerLimit = characteristicEntity.getLowerLimit();
+      double upperLimit = characteristicEntity.getUpperLimit();
       technicalLimits = Optional.of(new TechnicalLimits(lowerLimit, upperLimit));
     }
 
     Optional<MeanStddev> meanStddev = Optional.empty();
-    if (limitsEntity.getAutoAdjust()) {
-      double mean = limitsEntity.getComputedMean();
-      double stddev = limitsEntity.getComputedStddev();
+    if (characteristicEntity.getAutoAdjust()) {
+      int sampleSize = characteristicEntity.getSampleSize();
+      MeanStddevEntity meanStddevEntity;
+      if (characteristicEntity.getSampleSize() != null) {
```

Comment:
Nell'if sopra non hai messo `!= null`, qui si. Secondo me scegli uno stile e usa solo quello. Io di solito preferisco lasciare esplicito. A parte che ormai che hai assegnato SampleSize a un int in `int sampleSize = characteristicEntity.getSampleSize();` è sicuramente `not null`, altrimenti avrebbe sollevato un'eccezione.

Reply:
> Nell'if sopra non hai messo != null, qui si.

L'`if` sopra controlla se l'auto-adjust è `true`, non se sia `null`. L'auto-adjust non può essere `null`.

> A parte che ormai che hai assegnato SampleSize a un int in int sampleSize = characteristicEntity.getSampleSize();

Ops, quella variabile non è neanche utilizzata. L'ho tolta.

> altrimenti avrebbe sollevato un'eccezione.

Se converti un `Integer` a `int` il valore `null` viene convertito a 0 senza lanciare eccezioni.

Reply:
> Se converti un Integer a int il valore null viene convertito a 0 senza lanciare eccezioni.

Ah si? Forse era con gli OptionalInt che succedevano casini.

> L'if sopra controlla se l'auto-adjust è true, non se sia null. L'auto-adjust non può essere null.

Che palle, sempre sto casino che il getter deve chiamarsi `getSomething`, però da un bool mi aspetterei `is` o `has`. Così non era per niente intuitivo.